mantle/kola/tests/coretest/core.go

```diff
@@ -37,22 +43,45 @@ var offServices = []string{
 	"tcsd.service",
 }
 
+var nativeFuncs = map[string]register.NativeFuncWrap{
+	"PortSSH":        register.CreateNativeFuncWrap(TestPortSsh),
+	"DbusPerms":      register.CreateNativeFuncWrap(TestDbusPerms),
+	"ServicesActive": register.CreateNativeFuncWrap(TestServicesActive),
+	"ReadOnly":       register.CreateNativeFuncWrap(TestReadOnlyFs),
+	"Useradd":        register.CreateNativeFuncWrap(TestUseradd),
+	"MachineID":      register.CreateNativeFuncWrap(TestMachineID),
+	"RHCOSGrowpart":  register.CreateNativeFuncWrap(TestRHCOSGrowfs, []string{"fcos"}...),
+	"FCOSGrowpart":   register.CreateNativeFuncWrap(TestFCOSGrowfs, []string{"rhcos"}...),
+}
+
 func init() {
 	register.RegisterTest(&register.Test{
 		Name:        "basic",
 		Description: "Verify basic functionalities like SSH, systemd services, useradd, etc.",
 		Run:         LocalTests,
 		ClusterSize: 1,
-		NativeFuncs: map[string]register.NativeFuncWrap{
-			"PortSSH":        register.CreateNativeFuncWrap(TestPortSsh),
-			"DbusPerms":      register.CreateNativeFuncWrap(TestDbusPerms),
-			"ServicesActive": register.CreateNativeFuncWrap(TestServicesActive),
-			"ReadOnly":       register.CreateNativeFuncWrap(TestReadOnlyFs),
-			"Useradd":        register.CreateNativeFuncWrap(TestUseradd),
-			"MachineID":      register.CreateNativeFuncWrap(TestMachineID),
-			"RHCOSGrowpart":  register.CreateNativeFuncWrap(TestRHCOSGrowfs, []string{"fcos"}...),
-			"FCOSGrowpart":   register.CreateNativeFuncWrap(TestFCOSGrowfs, []string{"rhcos"}...),
-		},
+		NativeFuncs: nativeFuncs,
+	})
+	register.RegisterTest(&register.Test{
+		Name:        "basic.uefi",
+		Description: "Verify basic functionalities like SSH, systemd services, useradd, etc, with UEFI enabled",
+		Run:         uefiWithBasicTests,
+		ClusterSize: 0,
+		NativeFuncs: nativeFuncs,
+	})
+	register.RegisterTest(&register.Test{
+		Name:        "basic.uefi.secure",
+		Description: "Verify basic functionalities like SSH, systemd services, useradd, etc, with UEFI Secure Boot enabled",
+		Run:         uefiSecureWithBasicTests,
+		ClusterSize: 0,
+		NativeFuncs: nativeFuncs,
+	})
+	register.RegisterTest(&register.Test{
+		Name:        "basic.nvme",
+		Description: "Verify basic functionalities like SSH, systemd services, useradd, etc, with nvme enabled",
+		Run:         nvmeBasicTests,
+		ClusterSize: 0,
+		NativeFuncs: nativeFuncs,
```

Comment:
Wait sorry, I just noticed this. It's good now that we're no longer limiting these tests to x86_64, but we do still need to limit them to QEMU since that's the only place right now where we can directly control firmware and NVMe. `basic` test looks good as is.

Reply:
🤦 No need to apologize, that makes sense. I will do that real quick.